### PR TITLE
Remove duplicate AO momentum integral APIs

### PR DESCRIPTION
**Pull request by RTChou67**
Remove two redundant AO momentum integral APIs.
## Summary
  - Deleted the duplicate `qs_operators_ao::ang_mom` API and replaced it with the existing `ai_angmom::angmom` API.
  - Deleted the duplicate `qs_operators_ao::p_xyz_ao` API and replaced it with `qs_operators_ao::build_lin_mom_matrix`.
  - Extended `build_lin_mom_matrix` with the optional `minimum_image` argument.

### CODE REVIEW

File: src/qs_operators_ao.F [hunk 1]
```diff
@@ -121,6 +124,14 @@ SUBROUTINE build_lin_mom_matrix(qs_env, matrix)
       nkind = SIZE(qs_kind_set)
       natom = SIZE(particle_set)
 
+      my_minimum_image = .FALSE.
+      IF (PRESENT(minimum_image)) THEN
+         my_minimum_image = minimum_image
+         Lxo2 = SQRT(SUM(cell%hmat(:, 1)**2))/2.0_dp
+         Lyo2 = SQRT(SUM(cell%hmat(:, 2)**2))/2.0_dp
+         Lzo2 = SQRT(SUM(cell%hmat(:, 3)**2))/2.0_dp
```
Comment (fstein93) [inline src/qs_operators_ao.F]: Please use `NORM` instead.

File: src/qs_operators_ao.F [hunk 2]
```diff
@@ -185,6 +196,11 @@ SUBROUTINE build_lin_mom_matrix(qs_env, matrix)
          zetb => basis_set_b%zet
 
          IF (inode == 1) last_jatom = 0
+
+         IF (my_minimum_image) THEN
+            IF (ABS(rab(1)) > Lxo2 .OR. ABS(rab(2)) > Lyo2 .OR. ABS(rab(3)) > Lzo2) CYCLE
```
Comment (fstein93) [inline src/qs_operators_ao.F]: This can be written shorter: `ANY(ABS(rab(:)) > [Lxo2, Lyo2, Lzo2])`

**Review by fstein93 — CHANGES_REQUESTED** [review]

**Comment by RTChou67** [comment]
Also, could someone rerun the pdbg test? I believe that none of my commits are related to the slow tests.

**Review by fstein93 — APPROVED** [review]
